mcu/nanox_ocr.py

```diff
@@ -152,18 +151,15 @@ def analyze_bitmap(self, data: bytes):
 
         char = find_char_from_bitmap(bitmap)
         if char:
-            if self.text == b"":
-                self.x, self.y = (x, y)
-            elif y > self.last_y:
-                self.text += b"\n"
-            if char == "\x80":
-                char = " "
-            self.text += char.encode()
-
-        self.last_x, self.last_y = x, y
-
-    def get_text(self, reset=True) -> Tuple[bytes, Tuple[int, int]]:
-        text, x, y = self.text, self.x, self.y
-        if reset:
-            self.reset()
-        return text, (x, y)
+            # create a new TextEvent if there are no events yet or if there is a new line
+            if not self.events or y > self.events[-1].y:
+                event = TextEvent("", x, y)
+                self.events.append(event)
+            else:
+                event = self.events[-1]
+            event.text += char
```

Comment:
Or even simpler:
```python
            if not self.events or y > self.events[-1].y:
                self.events.append(TextEvent(char, x, y))
            else:
                self.events[-1].char += char
```

... which becomes even more readable by inversing the condition:

```python
            if self.events and y <= self.events[-1].y:
                self.events[-1].char += char
            else:
                self.events.append(TextEvent(char, x, y))
```